### ISSUE TITLE
What is a "custom size object" in easyprintMap

### ISSUE THREAD

**Issue by super-lou** [trafficonese/leaflet.extras2#31]
Hi, not a big js fan here ... I saw that it seems possible to add a [custom size object](https://github.com/rowanwins/leaflet-easyPrint#custom-print-sizes) as mention in the help of easyprintMap but I can't manage to make it works. Could you provide an example please ? 

Thank's by advance

**Comment by trafficonese** [comment]
Hey, I believe it might not be possible currently. I will also look into that. Thanks for opening the issue!

**Comment by super-lou** [comment]
Thank's! Even if the "Current" size option doesn't work I think that the possibility to have a custom size option could be the more interesting for the majority of people due to low resolution issue.
(Sorry not to be able to help more)

**Comment by trafficonese** [comment]
I just pushed [e84dcd1](https://github.com/trafficonese/leaflet.extras2/commit/e84dcd1d478059dd4e3ba18f05e14c8d96c56bad) which allows to use custom sizeModes. You can see an example here - [easyprint_app.R](https://github.com/trafficonese/leaflet.extras2/blob/master/inst/examples/easyprint_app.R)

The `CurrentSize` still only outputs gray background when used via the `easyprintMap` function. With the control buttons the `CurrentSize` options works fine and I have no idea why. It should be exactly the same code thats running..

**Comment by super-lou** [comment]
Wahou ! It works perfectly ... but not as i intended. 
It was definitly what I ask for but I don't know why I expect my screenshot to have better resolution... it obviously didn't change the dpi. Several issues are on the [leaflet-easyPrint](https://github.com/rowanwins/leaflet-easyPrint/issues/73) repo but no solution is given. If you have one I will definitly take it !
Otherwise I also looking for a solution for the `CurrentSize` option but thanks a lot for your work it is already a great help !

**Comment by trafficonese** [comment]
Yeah, I think the dpi-issue needs to happen in the dom-to-image library. There is a PR open right now that addresses it. https://github.com/tsayen/dom-to-image/pull/103/files

**Comment by super-lou** [comment]
If you think it can enable 300 dpi screenshot, that could really be a big help for me (and I think for anyone because 90 dpi is too low resolution today. ..). I'm sorry that I can't help you again with js but I completely believe in your ability to do so. Really hope this is possible !!

**Comment by trafficonese** [comment]
I just published a new branch [print_dpi](https://github.com/trafficonese/leaflet.extras2/tree/print_dpi), which allows to set the dpi.
You can see an example in [/inst/easyprint_app.R](https://github.com/trafficonese/leaflet.extras2/blob/print_dpi/inst/examples/easyprint_app.R)

It seems to work, but it only really affects vector-layer, labels and popups. The basemap will not be in a higher resolution.

**Comment by super-lou** [comment]
It works great and I think it depends on the basemap! Personally, I really see a big improvement.

90 dpi :
![map_90](https://user-images.githubusercontent.com/60742732/162741413-a7bafd09-5031-41c6-9424-98e779188cde.png)
300 dpi :
![map_300](https://user-images.githubusercontent.com/60742732/162741430-6f7ef05e-5e0c-4f79-b37b-4aa748970c34.png)

I'm really happy with the possibility you're giving me and everything else will be bonus! But I can still ask for more haha.
Is it possible to export it in pdf (and thus to keep the vectorized shapes of leaflet and perhaps if it exists of the basemap)?

**Comment by Nova-Scotia** [comment]
- I've been testing this out and at least with the ESRI provider tiles I can't seem to get the labels to "crisp up" - frustrating because they *are* crisp when rendered by `leaflet!`

**Comment by lou-heraut** [comment]
That’s why it would be great to have a PDF export option, but in three years I haven’t had the opportunity to dig into that topic. For your information, I stopped developing in Shiny partly because of this issue, even in JavaScript, exporting figures hasn’t been a straightforward problem in my experience.
